schema.graphql

```diff
@@ -1,67 +1,25 @@
-type TX @entity {
-  id: ID!
-  createdAt: String!
-}
-# TOP HATS MAY BE UNNECESSARY
-# Entity that recursively maps an TopHat's entire tree
-type TopHatTree @entity {
-  id: ID!
-  createdAt: String!
-  owner: Bytes!
-  subHats: [HatTree!]
-}
-# Finite TopHat data
-type TopHat @entity {
-  id: ID!
-  createdAt: String!
-  owner: Bytes!
-  subHats: [HatTree!]
-}
-# Entity that recursively maps an Hat's tree
-type HatTree @entity {
-  id: ID!
-  createdAt: String!
-  name: String
-  description: String
-  imageUrl: String
-  # wearer: [WearerTransferReceipt!]
-  subHats: [HatTree!]
-}
-# Finite Hat data
 type Hat @entity {
```

Comment:
Could also include the following attributes for the Hat entity, subject to cost-benefit analysis of including in the subgraph vs on the front end

- admin (retrievable from the id, but might be handy to have concretely, especially in combination with the removed HatTree entity -- eg to look up sibling hats)
- current supply (length of the wearers array)

Reply:
What are your thoughts on also keeping track of changes to hat properties?

Reply:
Oh, the other thing that would be amazing to have here would be a "pretty id", which formats an id kind of like an IP address...
- in hex, with the `0x` prefix
- periods between the hat levels
- only show the levels with non-zero values

For example, the pretty id of a level 3 hat might look like this: `00000001.01.01`

Reply:
Maybe also a history of wearers of the hat?